From e6d23098d4f91a4e7dd88a9a9a028cc99da7ae17 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Thu, 23 Nov 2023 08:51:56 +0700
Subject: [PATCH] Ignore using OpenSSL's libcrypto library for host tools

When compiling u-boot with Jessie CIP_MODE (openssl v1.0.1t), below
errors are occured:

| lib/ecdsa/ecdsa-libcrypto.c: In function 'alloc_ctx':
| lib/ecdsa/ecdsa-libcrypto.c:39:7: warning: implicit declaration of function 'OPENSSL_i
nit_ssl' [-Wimplicit-function-declaration]
|   if (!OPENSSL_init_ssl(0, NULL)) {
|        ^
| lib/ecdsa/ecdsa-libcrypto.c: In function 'ecdsa_sig_encode_raw':
| lib/ecdsa/ecdsa-libcrypto.c:77:2: warning: implicit declaration of function 'ECDSA_SIG
_get0' [-Wimplicit-function-declaration]
|   ECDSA_SIG_get0(sig, &r, &s);
|   ^
| lib/ecdsa/ecdsa-libcrypto.c:79:2: warning: implicit declaration of function 'BN_bn2bin
pad' [-Wimplicit-function-declaration]
|   BN_bn2binpad(r, buf, point_bytes);
|   ^
| lib/ecdsa/ecdsa-libcrypto.c: In function 'ecdsa_sig_from_raw':
| lib/ecdsa/ecdsa-libcrypto.c:98:2: warning: implicit declaration of function 'ECDSA_SIG
_set0' [-Wimplicit-function-declaration]
|   ECDSA_SIG_set0(sig, r, s);
|   ^
| lib/ecdsa/ecdsa-libcrypto.c: In function 'ecdsa_key_size_bytes':
| lib/ecdsa/ecdsa-libcrypto.c:109:9: warning: implicit declaration of function 'EC_GROUP
_order_bits' [-Wimplicit-function-declaration]
|   return EC_GROUP_order_bits(group) / 8;
|          ^
| lib/ecdsa/ecdsa-libcrypto.c: In function 'do_add':
| lib/ecdsa/ecdsa-libcrypto.c:288:2: warning: implicit declaration of function 'EC_POINT
_get_affine_coordinates' [-Wimplicit-function-declaration]
|   EC_POINT_get_affine_coordinates(group, point, x, y, NULL);
|   ^

These functions are supported from openssl v1.1.0.
So ignore using OpenSSL's libcrypto library for host tools when building
with Jessie openssl version.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 configs/hihope_rzg2_defconfig   | 1 +
 configs/silinux_ek874_defconfig | 1 +
 tools/Makefile                  | 1 -
 3 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/configs/hihope_rzg2_defconfig b/configs/hihope_rzg2_defconfig
index 9f87ac74f4f..f00252620ba 100644
--- a/configs/hihope_rzg2_defconfig
+++ b/configs/hihope_rzg2_defconfig
@@ -79,3 +79,4 @@ CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_WDT=y
 CONFIG_WDT_RENESAS=y
 CONFIG_CMD_WDT=y
+CONFIG_TOOLS_LIBCRYPTO=n
diff --git a/configs/silinux_ek874_defconfig b/configs/silinux_ek874_defconfig
index 4473092cc09..08f5b378c1a 100644
--- a/configs/silinux_ek874_defconfig
+++ b/configs/silinux_ek874_defconfig
@@ -76,3 +76,4 @@ CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_WDT=y
 CONFIG_WDT_RENESAS=y
 CONFIG_CMD_WDT=y
+CONFIG_TOOLS_LIBCRYPTO=n
diff --git a/tools/Makefile b/tools/Makefile
index 4a86321f646..158b3ade4dd 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -117,7 +117,6 @@ dumpimage-mkimage-objs := aisimage.o \
 			imximage.o \
 			imx8image.o \
 			imx8mimage.o \
-			kwbimage.o \
 			lib/md5.o \
 			lpc32xximage.o \
 			mxsimage.o \
-- 
2.17.1

